components/button_selector.jsx

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class ButtonSelector extends React.PureComponent {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        value: PropTypes.string.isRequired,
+        onChange: PropTypes.func,
+        label: PropTypes.node,
+        labelClassName: PropTypes.string,
+        inputClassName: PropTypes.string,
+        helpText: PropTypes.node,
+        footer: PropTypes.node,
+        disabled: PropTypes.bool,
+        shouldSubmit: PropTypes.bool,
+        options: PropTypes.array,
+    };
+
+    static defaultProps = {
+        value: '',
+        labelClassName: '',
+        inputClassName: '',
+    };
+
+    onClick = (value) => {
+        if (this.props.onChange) {
+            this.props.onChange(this.props.id, value);
+        }
+    }
+
+    render() {
+        const {
+            footer,
+            label,
+            labelClassName,
+            helpText,
+            inputClassName,
+            disabled,
+            options,
+            shouldSubmit,
+            value,
+        } = this.props;
+
+        let labelContent;
+        if (label) {
+            labelContent = (
+                <label
+                    className={'control-label ' + labelClassName}
+                >
+                    {label}
+                </label>
+            );
+        }
+
+        let helpTextContent;
+        if (helpText) {
+            helpTextContent = (
+                <div className='help-text'>
+                    {helpText}
+                </div>
+            );
+        }
+
+        const buttons = options.map((opt) => {
+            let className = 'btn btn-link';
+            if (disabled) {
+                className += ' btn-inactive';
+            }
+            if (opt.value === value) {
+                className += '  btn-primary';
+            }
+            const type = shouldSubmit ? 'submit' : 'button';
+            return (
+                <button
+                    key={opt.value}
+                    type={type}
```

Comment:
Can you describe the difference in behavior between `type='button'` and `type='submit'` here? I feel like it would make more sense for the parent component to be aware of when a certain button is pressed to submit, and then handle the submission itself. Rather than the form being submitted through a native browser form submit event.

As it currently stands: When the submit button is clicked, is the `onClick` handler called before the form is submitted?

Reply:
When the submit button is clicked, it first trigger the onClick (this allow for other elements to do things like `e.preventDefault()`).
The reason to do it this way is because the submission of the form (the send button) does not apply any other logic, it just submits the form (which is the one using the `handleSubmit` function). I think this is the cleanest way to do it.

Reply:
One thing that comes to mind is that pressing enter while typing in one of the inputs could trigger one of the buttons to submit, probably the uppermost submit button in the form. If this occurs, do you think that is an issue?

Reply:
It may become an issue, but I feel if that issue exist, we already have it.
The main reason it would be an issue is because, specially for autocomplete, we may hit enter to select the autocomplete field we want. I will test it and create an issue if needed.